Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -23,6 +23,7 @@ ComponentInfo::ComponentInfo(
     boost::shared_ptr<const std::vector<size_t>> parentIndices,
     boost::shared_ptr<std::vector<Eigen::Vector3d>> positions,
     boost::shared_ptr<std::vector<Eigen::Quaterniond>> rotations,
+    boost::shared_ptr<std::vector<Eigen::Vector3d>> scaleFactors,
```

Comment:
Very few things use the scale factors. Doesn't this mean that `ComponentInfo` will require a set of scale factors to be constructed?